src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -7,44 +7,91 @@
 
 package frc.robot.subsystems;
 
-import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 import com.revrobotics.CANPIDController;
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.CANSparkMaxLowLevel;
 import com.revrobotics.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
 
 import frc.robot.Constants;
 
 public class ShooterSubsystem extends SubsystemBase {
-  private final CANSparkMax ShooterMAX = new CANSparkMax(Constants.SPARKMAX_SHOOTER_CAN_ADDR,
+  /**
+   * Initializes the SparkMAX motor controller, assigns it to the CAN address
+   * specified, and sets it to the NEO Brushless Motor.
+   */
+  private final CANSparkMax shooterMAX = new CANSparkMax(Constants.SPARKMAX_SHOOTER_CAN_ADDR,
       CANSparkMaxLowLevel.MotorType.kBrushless);
-
-  private final CANPIDController ShooterPID = new CANPIDController(ShooterMAX);
-
-  private final Encoder shootEncoder = new Encoder(Constants.SHOOTER_ENCODER_CHAN_A, Constants.SHOOTER_ENCODER_CHAN_B);
-
-  private final double ShooterSetpoint = 1.0;
-
+  /**
+   * The built-in PID controller provided by the Spark MAX motor controller.
+   */
+  private final CANPIDController shooterPID = new CANPIDController(shooterMAX);
+  /**
+   * The target velocity of the NEO Brushless Motor.
+   */
+  private final double shooterSetpoint = 1.0;
+  /**
+   * The Proportial Gain of the SparkMAX PIDF controller The weight of the
+   * proportional path against the differential and integral paths is controlled
+   * by this value.
+   */
   private final double kP = 1;
```

Comment:
Research default PID constants for the NEO.  You'll find that they're much smaller than 1.  Setting them to 1 can be surprisingly violent.  (Ask Jake about his laptop)